File: internal/api/projects.go
package api

import "fmt"

// ListProjects returns a paginated list of projects.
func ListProjects(c *Client, query map[string]string) ([]map[string]interface{}, *PaginationMeta, error) {
	var resp struct {
		Projects []map[string]interface{} `json:"projects"`
		Meta     *PaginationMeta          `json:"meta"`
	}
	if err := c.Get("/api/public/projects", query, &resp); err != nil {
		return nil, nil, err
	}
	return resp.Projects, resp.Meta, nil
}

// GetProject returns a single project by ID.
func GetProject(c *Client, id string) (map[string]interface{}, error) {
	var resp map[string]interface{}
	if err := c.Get(fmt.Sprintf("/api/public/projects/%s", id), nil, &resp); err != nil {
		return nil, err
	}
	if proj, ok := resp["project"].(map[string]interface{}); ok {
		return proj, nil
	}
	return resp, nil
}

File: python/stqry/projects.py
class ProjectsResource:
    def __init__(self, http):
        self._http = http

    def list(self, **query) -> list:
        resp = self._http.get("/api/public/projects", params=query or None)
        return resp.get("projects", [])

    def get(self, id: str) -> dict:
        resp = self._http.get(f"/api/public/projects/{id}", params=None)
        return resp.get("project", resp)
